qiita_db/sql_connection.py

```diff
@@ -22,57 +22,139 @@
-conn_handler.execute_fetchall(
-    "SELECT * from qiita.qiita_user WHERE email = %s",
-    ['insert@foo.bar']) # doctest: +SKIP
-[['insert@foo.bar', 1, 'pass', 'Toy', None, None, '222-222-2221', None, None,
-  None]] # doctest: +SKIP
-
-You can also use results from a previous command in the queue in a later
-command. If an item in the queue depends on a previous sql command's output,
-use {#} notation as a placeholder for the value. The \# must be the
-position of the result, e.g. if you return two things you can use \{0\}
-to reference the first and \{1\} to referece the second. The results list
-will continue to grow until one of the references is reached, then it
-will be cleaned out.
-Modifying the previous example to show this ability (Note the RETURNING added
-to the first SQL command):
-
-from qiita_db.sql_connection import SQLConnectionHandler
-conn_handler = SQLConnectionHandler # doctest: +SKIP
-conn_handler.create_queue("example_queue") # doctest: +SKIP
-conn_handler.add_to_queue(
-    "example_queue", "INSERT INTO qiita.qiita_user (email, name, password,"
-    "phone) VALUES (%s, %s, %s, %s) RETURNING email, password",
-    ['insert@foo.bar', 'Toy', 'pass', '111-111-11112']) # doctest: +SKIP
-conn_handler.add_to_queue(
-    "example_queue", "UPDATE qiita.qiita_user SET user_level_id = 1, "
-    "phone = '222-222-2221' WHERE email = %s AND password = %s",
-    ['{0}', '{1}']) # doctest: +SKIP
-conn_handler.execute_queue("example_queue") # doctest: +SKIP
-conn_handler.execute_fetchall(
-    "SELECT * from qiita.qiita_user WHERE email = %s", ['insert@foo.bar'])
-[['insert@foo.bar', 1, 'pass', 'Toy', None, None, '222-222-2221', None, None,
-  None]] # doctest: +SKIP
+
+>>> from qiita_db.environment_manager import drop_and_rebuild_tst_database
+>>> drop_and_rebuild_tst_database()  # doctest: +SKIP
+
+* Querying
+
+In order to perform any query in the database you first need to import the
+`TRN` variable available in this module:
+
+>>> from qiita_db.sql_connection import TRN
+
+The `TRN` variable is an instance of the Transaction object. All queries should
+be executed through this transaction object to avoid committing anything to the
+database that is partially executed. To perform any query you first need to
+add the query to the transaction and then execute the transaction:
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     res = TRN.execute()
+>>> res
+[[[42]]]
+
+* Data retrieval
+
+Multiple auxiliary functions exist for easy SQL result retrieval:
+
+Getting the first value of the last SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchlast()
+>>> res
+42
+
+Getting the results of the last SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchindex()
+>>> res
+[[42]]
+
+Getting the results of the specified SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 43")
+...     res = TRN.execute_fetchindex(0)
```

Comment:
It would be nice to explicitly note that the value `0` here corresponds to the first query in the transaction